tests/cell_test.cc

```diff
@@ -223,11 +223,15 @@ TEST_CASE("Cell is checked for 2D case", "[cell][2D]") {
         point << 0., 0.;
         REQUIRE(cell->point_in_cartesian_cell(point) == true);
         REQUIRE(cell->is_point_in_cell(point, &xi) == true);
+        REQUIRE(xi(0) == -1. + std::numeric_limits<double>::epsilon());
+        REQUIRE(xi(1) == -1. + std::numeric_limits<double>::epsilon());
```

Comment:
Comparing doubles this way may not work all the time, we may have to do something like this?

```
template<class T>
typename std::enable_if<!std::numeric_limits<T>::is_integer, bool>::type
    almost_equal(T x, T y, int ulp)
{
    // the machine epsilon has to be scaled to the magnitude of the values used
    // and multiplied by the desired precision in ULPs (units in the last place)
    return std::fabs(x-y) <= std::numeric_limits<T>::epsilon() * std::fabs(x+y) * ulp
        // unless the result is subnormal
        || std::fabs(x-y) < std::numeric_limits<T>::min();
}
```

or alternatively, we could use the `Approx` call in `REQUIRE` macro with `epsilon` to check if the results are correct.

Reply:
Thanks for the suggestion, I decided to use `Approx` with machepsilon. Though, this also allows the value to be machepsilon different from the provided value. In our example, this will return true if `xi(i) = -1`. That's the reason I initially did doubles comparison.